### PR TITLE
[P8-99] Fix clippy warnings

### PR DESCRIPTION
**Pull request by lars4509**
I have tried to fix multiple clippy warnings. One of the main issues is the naming conventions which have been easily fixed, following [https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case](url).

### CODE REVIEW

**Comment by lars4509** [comment]
I have fixed most of the warnings, and ignored only a few. If you are able to figure out a solution for the one i have ignored, that would be great!

File: atl-checker/src/game_structure/lcgs/ir/symbol_checker.rs [hunk 1]
```diff
@@ -243,6 +243,7 @@ impl<'a> SymbolChecker<'a> {
     }
     /// First combines all numbers, as we already know the min of that
     /// Then returns a new checked Vec of Expr to find Min of.
+    #[allow(clippy::unnecessary_unwrap)]
```
Comment (NicEastvillage) [inline atl-checker/src/game_structure/lcgs/ir/symbol_checker.rs:246]:
```suggestion
    #[allow(clippy::unnecessary_wraps)]
```

File: atl-checker/src/game_structure/lcgs/ir/symbol_checker.rs [hunk 2]
```diff
@@ -277,6 +278,7 @@ impl<'a> SymbolChecker<'a> {
     }
     /// First combines all numbers, as we already know the max of that
     /// Then returns a new checked Vec of Expr to find Max of.
+    #[allow(clippy::unnecessary_unwrap)]
```
Comment (NicEastvillage) [inline atl-checker/src/game_structure/lcgs/ir/symbol_checker.rs:281]:
```suggestion
    #[allow(clippy::unnecessary_wraps)]
```

**Review by NicEastvillage — CHANGES_REQUESTED** [review]
There are some missing renames here:
- `atlchecker::algorithms::certain_zero:common::VertexAssignment`
- `atlchecker::algorithms::certain_zero:common::Message`
- `cli::FormulaFormat`
- `cli::ModelType`
- `cli::SearchStrategyOption`

File: atl-checker/src/game_structure/lcgs/ir/symbol_checker.rs (hunk 2 above)
Comment (lars4509) [reply atl-checker/src/game_structure/lcgs/ir/symbol_checker.rs:281]: I have tried` #[allow(clippy::unnecessary_wraps)]` but i am getting this warning.

image

**Comment by lars4509** [comment]
> There are some missing renames here:
> 
> * `atlchecker::algorithms::certain_zero:common::VertexAssignment`
> * `atlchecker::algorithms::certain_zero:common::Message`
> * `cli::FormulaFormat`
> * `cli::ModelType`
> * `cli::SearchStrategyOption`

It seems like my clippy doesn't catch these for some reason, but i will fix those as well.

File: atl-checker/src/game_structure/lcgs/ir/symbol_checker.rs (hunk 2 above)
Comment (NicEastvillage) [reply atl-checker/src/game_structure/lcgs/ir/symbol_checker.rs:281]: Huh. But `unwrap` doesn't remove the clippy warning either. I guess, just skip this one then.

Comment (lars4509) [reply atl-checker/src/game_structure/lcgs/ir/symbol_checker.rs:281]: After i updated my clippy, it seems like it recognize `clippy:unnecessary_wraps` and the warning doesn't appear for me anymore.

**Comment by lars4509** [comment]
After an update to rustup, my clippy seems to catch these warnings now. They should have been fixed now.

> > There are some missing renames here:
> > 
> > * `atlchecker::algorithms::certain_zero:common::VertexAssignment`
> > * `atlchecker::algorithms::certain_zero:common::Message`
> > * `cli::FormulaFormat`
> > * `cli::ModelType`
> > * `cli::SearchStrategyOption`
> 
> It seems like my clippy doesn't catch these for some reason, but i will fix those as well.

**Review by NicEastvillage — APPROVED** [review]
Awesome! Can't wait for all the errors/merge conflicts this will produce

**Review by falkecarlsen — APPROVED** [review]

**Pull request merged**